### ISSUE TITLE
Buffering issue in main()

### ISSUE THREAD

**Issue by dandelionred** [grantjenks/python-wordsegment#12]
I wanted to use wordsegment from php as a by-line processor so `python -m wordsegment` gonna be all I need. But it is completely not usable because of this piece in `main()`:
```python
    for line in streams.infile:
        streams.outfile.write(' '.join(segment(line)))
        streams.outfile.write(os.linesep)
```

`stdin.infile`'s buffering comes into play and I basicly don't get any response after feeding a line into it:
```
$ sudo strace -p $(pidof python)
Process 12343 attached
read(0, "thisisatest\n", 8192)          = 12
read(0,
```

See it gets the line and waits for more input because `stdin.infile`'s buffer has not been saturated yet. Btw the buffering can't be disabled with python's `-u` option.

Please consider remaking the `main()` code so by-line processing works out of the box for `python -m wordsegment` :+1:

**Comment by grantjenks** [comment]
Do you mean that you tried the -u option and that it did not work or you cannot try the -u option in your environment? I thought that's what it's designed for.

I think there is also an environment variable you can set to disable buffering. Would that work in your environment?

**Comment by dandelionred** [comment]
python man says it explicitly sys.stdin iterator has its own buffering
```
       -u     Force  stdin,  stdout  and  stderr to be totally unbuffered.  On
              systems where it matters, also put stdin, stdout and  stderr  in
              binary  mode.   Note  that there is internal buffering in xread‐
              lines(), readlines() and file-object  iterators  ("for  line  in
              sys.stdin")  which  is  not  influenced by this option.  To work
              around this, you will want to use "sys.stdin.readline()"  inside
              a "while 1:" loop.
```

As for env vars PYTHONUNBUFFERED=1 is the same as '-u'.

So there is no quick fix for this issue.

I guess in `main()` you could iterate like this instead:
```python
	line = sys.stdin.readline()
	while line:
		print(' '.join(wordsegment.segment(line)))
		line = sys.stdin.readline()
```

Optionally add `sys.stdout.flush()` after `print()`. Optionally because you can achieve per-line flushes with `stdbuf -oL python ...` (or completely unbuffered writes with `python -u ...` or `stdbuf -o0 python ...`).

**Comment by grantjenks** [comment]
Ok, thanks for researching the issue. Stdin should use readline in the while-loop as you describe. Although I think iter(says.stdin, '') would work better in a for-loop.

I would like to leave stdout as-is. If you make a pull request, I will merge and release. I won't get around to making this change myself until the end of the month.

**Comment by dandelionred** [comment]
I'm a seasonal user of python so I doubt my code is suitable for public :)
As for myself I've made a line-iterating wrapper around wordsegment so I'm OK.

**Comment by grantjenks** [comment]
Fixed and deployed at v1.1.3 or greater.
